modules/geolocation/geolocation.py

```diff
@@ -585,11 +599,23 @@ def run_locator(self, telemetry, coordinates):
         if len(point_pairs) < 4:
             return False, None
 
-        non_collinear_points = self.get_non_collinear_points(point_pairs)
+        # Slice point_pairs to shape (n, 2) for input of get_non_collinear_points
+        points = point_pairs[:,1]
+
+        # Get the 4 non-collinear indexes of the array above
+        indexes = self.get_non_collinear_points(points)
+
         # If insufficient point pairs, exit this run and try again
-        if len(non_collinear_points) < 4:
+        if len(indexes) < 4:
             return False, None
 
+        non_collinear_points = np.array(indexes)
+
+        # Create a subset of the (n, 2, 2) array above using the array of indexes
+        non_collinear_points = [point_pairs[i] for i in indexes]
+        # non_collinear_points only stores the 4 non-collinear point pairs
+        # indicated by the indexes array
+
```

Comment:
Sorry, what I meant is that you can take the subset from an array using the square bracket `[]` operator, except instead of of a number inside, you can put an array and it will take the elements.

Example:
```python
a = np.array([2, 4, 6, 8])
b = np.array([0, 3])
print(a[b]) # Outputs: [2 8]
```